### PR TITLE
feat(push-server): initial import of push server from 5c84bb95c121091981e3b44d0f61cf85e2b9d8a1

### PR DESCRIPTION
**Pull request by bassrock**
## Goal

Move the pocket instant sync [push server ](https://github.com/Pocket/PushServer)

https://mozilla-hub.atlassian.net/browse/POCKET-9451

## Todos
- [x] Import code
- [x] refactor push code to our latest standards
- [x] redo infra to match latest patterns (this is a background process and is safe to do so)

### CODE REVIEW

File: infrastructure/push-server/src/main.ts [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+      bucket: `mozilla-pocket-team-${config.environment.toLowerCase()}-terraform-state`,
+      dynamodbTable: `mozilla-pocket-team-${config.environment.toLowerCase()}-terraform-state`,
+      key: config.name,
+      region: 'us-east-1',
+    });
+
+    const region = new DataAwsRegion(this, 'region');
+    const caller = new DataAwsCallerIdentity(this, 'caller');
+
+    //NOTE: THis service uses CPU based autoscaling, this should move to SQS based autoscaling based on the Job queue in the future.
```
Comment (bassrock) [inline infrastructure/push-server/src/main.ts:48]: Looking at its past scaling events, converting this to be CPU bound scaling should be fine for a while. This service has backed up before and handled through the messages.

File: servers/push-server/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import { Worker, isMainThread } from 'worker_threads';
+import Sentry from './sentry';
+import { worker } from './worker';
+import { numWorkers, msBetweenStarts } from './config';
+
+if (isMainThread) {
```
Comment (bassrock) [inline servers/push-server/src/index.ts:6]: we could replace all this worker logic with https://github.com/bbc/sqs-consumer and really clean up the logic

File: infrastructure/push-server/src/main.ts (hunk 1 above)
Comment (bassrock) [reply infrastructure/push-server/src/main.ts:48]: Once we have this, we could convert some of the SQS lambdas to ecs tasks which are easier to test and manage too.

File: servers/push-server/README.md
Comment (kschelonka) [inline servers/push-server/README.md:1]: https://mozilla-hub.atlassian.net/wiki/spaces/PE/pages/390629672/Push+Notifications

**Review by kschelonka — APPROVED** [review]

**Pull request merged**